src/Aquifer.Public.API/Endpoints/Resources/Get/AvailableLanguages/Endpoint.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Aquifer.Data;
+using FastEndpoints;
+using Microsoft.EntityFrameworkCore;
+
+namespace Aquifer.Public.API.Endpoints.Resources.Get.AvailableLanguages;
+
+public class Endpoint(AquiferDbContext dbContext) : Endpoint<Request, IEnumerable<Response>>
+{
+    public override void Configure()
+    {
+        Get("/resources/{ContentId}/available-languages");
+        Description(d => d.ProducesProblemFE(404));
+        Summary(s =>
+        {
+            s.Summary = "For a given resource content id, see in what other languages it is available.";
+            s.Description = "For a given resource content id, return a list of all other languages that are available for this resource.";
+        });
+    }
+
+    public override async Task HandleAsync(Request req, CancellationToken ct)
+    {
+        var foundContent = await dbContext.ResourceContentVersions.Where(x => x.ResourceContentId == req.ContentId && x.IsPublished)
+            .Select(x => new ResourceContentQueryResponse(x.ResourceContent.ResourceId, x.ResourceContent.LanguageId))
+            .FirstOrDefaultAsync(ct);
+
+        if (foundContent is null)
+        {
+            await SendNotFoundAsync(ct);
+            return;
+        }
+
+        Response = await dbContext.ResourceContentVersions
+            .Where(x => x.ResourceContent.ResourceId == foundContent.ResourceId &&
+                x.IsPublished &&
+                x.ResourceContent.LanguageId != foundContent.LanguageId)
+            .Select(x => new Response
+            {
+                ContentId = x.ResourceContentId,
+                ContentDisplayName = x.DisplayName,
+                LanguageId = x.ResourceContent.LanguageId,
+                LanguageDisplayName = x.ResourceContent.Language.DisplayName,
+                LanguageCode = x.ResourceContent.Language.ISO6393Code,
+                LanguageEnglishDisplayName = x.ResourceContent.Language.EnglishDisplay
+            })
+            .ToListAsync(ct);
```

Comment:
Could we simplify this to one query and return the 404 if the response list is empty?

Reply:
I was struggling to come up with a good way to do it, and I don't want to send back the content id that's provided. If you have an idea that runs faster I'm open to it, but this is at least pretty clean and clear what's going on and I don't think the performance is bad.

Reply:
Also, if the list is empty, implying no other languages were found, I don't think it's correct to send back a 404. I see the 404 as being the content id wasn't found at all, whereas an empty list is a 200 and indicates there are simply no other available languages.

Reply:
Could you:
- pull the list (including current language)
- if empty 404
- filter out given content id
- respond

If it's not trivial to switch though then it's not worth it, I just thought it could improve it slightly.

Reply:
I think it'll create a weird query that won't perform any better and isn't worth it personally.